ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeServiceBackend.scala

```diff
@@ -53,4 +56,11 @@ object TimeServiceBackend {
           else Future.successful(false)
         }(DirectExecutionContext)
   }
+
+  private class WallClockTimeServiceBackend(clock: Clock) extends TimeServiceBackend {
+    override def getCurrentTime: Instant = clock.instant()
+
+    override def setCurrentTime(currentTime: Instant, newTime: Instant): Future[Boolean] =
+      Future.failed(new IllegalArgumentException("WallClockTimeServiceBackend is read-only"))
```

Comment:
```suggestion
      Future.failed(new UnsupportedOperationException("WallClockTimeServiceBackend is read-only"))
```

Reply:
That code is not needed anymore but thanks!